Sources/NetworkProtection/Diagnostics/NetworkProtectionTunnelFailureMonitor.swift

```diff
@@ -100,6 +102,14 @@ public actor NetworkProtectionTunnelFailureMonitor {
     // MARK: - Handshake monitor
 
     private func monitorHandshakes(callback: @escaping (Result) -> Void) async {
+        guard firstCheckSkipped else {
+            // Avoid running the first tunnel failure check after startup to avoid reading the first handshake after sleep, which will almost always
+            // be out of date. In normal operation, the first check will frequently be 0 as WireGuard hasn't had the chance to handshake yet.
+            os_log("⚫️ Skipping first tunnel failure check", log: .networkProtectionTunnelFailureMonitorLog, type: .debug)
+            firstCheckSkipped = true
+            return
+        }
```

Comment:
This addresses the case where the tunnel monitor is checking too soon after sleep